mu/interface/dialogs.py

```diff
@@ -43,12 +41,30 @@
 )
 from PyQt5.QtGui import QTextCursor
 from mu.resources import load_icon
-from mu.logic import MODULE_DIR
+
 from mu.interface.widgets import DeviceSelector
 
 logger = logging.getLogger(__name__)
 
 
+class MuWidget(QWidget):
```

Comment:
Thanks, @dybber. Basically, I think you're right and I'll have a go at a branch this morning to see how straightforward a disentangle that would be.

Obviously this change has come by a long road (pretty much a year since I took it on) and it's definitely good to take a step back after all the twists and turns and see if there isn't a straighter path. The problem was that the need for a venv kept popping up way down the stack inside dialog boxes etc. and this seemed like a way to make it available.

For some reason, the idea of a singleton venv (which is what it basically is) never really occurred to me. Wood and Trees and that kind of thing.